lib/oidcstrategy.js

```diff
@@ -506,13 +585,11 @@ Strategy.prototype.authenticate = function authenticateStrategy(req, options) {
 
           // if (policy) { params['p'] = policy; }; // Policy parameter should be included.
 
-          // Add support for automatically generating a random state for verification.
-
-          let state;
-
-          if (!options.state) { state = utils.uid(16); } else { state = options.state; }
-
-          params.state = state;
+          // Add an automatically generated state to params, and remember it.
+          // Later when we get the id_token or authorization code back, we can 
+          // verify the state we sent are the same as the state returned in the
+          // query or body.
+          saved_state = params.state = utils.uid(16);
```

Comment:
Is `params` intended to be an input to this module, an output, or both? When I first looked at it, I had assumed that I could pass in the value of `state` when I called this module to 'override' the state generation with my own mechanism. Whereas this line implies that you're overwriting whatever value of 'state' is passed in.

I don't personally need this feature, but it does seem like a point of confusion.